pyfakefs/fake_filesystem.py

```diff
@@ -1994,24 +2059,26 @@ def get_object_from_normpath(self, file_path, check_read_perm=True):
 
         file_path = self._original_path(file_path)
         path_components = self._path_components(file_path)
-        target_object = self.root
+        target = self.root
         try:
             for component in path_components:
-                if S_ISLNK(target_object.st_mode):
-                    target_object = self.resolve(target_object.contents)
-                if not S_ISDIR(target_object.st_mode):
+                if S_ISLNK(target.st_mode):
+                    if target.contents:
+                        target = self.resolve(target.contents)
+                if not S_ISDIR(target.st_mode):
                     if not self.is_windows_fs:
                         self.raise_os_error(errno.ENOTDIR, file_path)
                     self.raise_os_error(errno.ENOENT, file_path)
-                target_object = target_object.get_entry(component)
-                if (not is_root() and check_read_perm and target_object and
-                        not target_object.st_mode & PERM_READ):
-                    self.raise_os_error(errno.EACCES, target_object.path)
+                target = target.get_entry(component)  # type: ignore
```

Comment:
Incompatible types in assignment (expression has type "Union[FakeFile, FakeDirectory]", variable has type "FakeDirectory") 
Argument 1 to "get_entry" of "FakeDirectory" has incompatible type "bytes"; expected "str"